### ISSUE TITLE
segmentation fault with bun run 😱😰

### ISSUE THREAD

**Issue by subtleGradient** [oven-sh/bun#9748]
### What version of Bun is running?

1.0.36

### What platform is your computer?

Darwin 23.0.0 arm64 arm

### What steps can reproduce the bug?

1. check out this code: https://github.com/subtleGradient/rsc-fun-in-bun/tree/0b38f7325d6d0429681282e3f060f3780e49fb01
2. `bun i`
3. `bun run ./demo-RSC.server.tsx`
4. `open 'http://localhost:3000/'`

segmentation fault 😰

```
% bun run ./demo-RSC.server.tsx
Server running at http://localhost:3000/
zsh: segmentation fault  bun run ./demo-RSC.server.tsx
```

### What is the expected behavior?

I expect it to not segmentation fault

### What do you see instead?

it segmentation faults

### Additional information

🤷‍♂️

**Comment by subtleGradient** [comment]
I also had to reboot my computer after `bun run --watch ./demo-RSC.server.tsx` became an unkillable zombie process monopolizing port 3000 forever. Probably related

**Comment by subtleGradient** [comment]
also got this, not sure if this helps

```
% bun run --inspect ./demo-RSC.server.tsx
--------------------- Bun Inspector ---------------------
Listening:
  ws://localhost:6499/c0qn1h52x64
Inspect in browser:
  https://debug.bun.sh/#localhost:6499/c0qn1h52x64
--------------------- Bun Inspector ---------------------
Server running at http://localhost:3000/
Thread 0x1e0929300 encountered bitfit alloaction error.
Bits for page 0x104f24230 (bmalloc_marge_bitfit):
free bits: 000000000000000000001111111000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000010000000000000000000000000000000000000001111111111111111111111111000000000000000000000000000000011111111111111111111111100000000000000000000000000000000000000000000000000000000000000000000000000000111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111zsh: segmentation fault  bun run --inspect ./demo-RSC.server.tsx
```

**Comment by subtleGradient** [comment]
at 0b38f7325d6d0429681282e3f060f3780e49fb01 here: https://github.com/subtleGradient/rsc-fun-in-bun/blob/0b38f7325d6d0429681282e3f060f3780e49fb01/demo-RSC.server.tsx

Now I'm seeing this:

```
% bun run --inspect ./demo-RSC.server.tsx
--------------------- Bun Inspector ---------------------
Listening:
  ws://localhost:6499/4hki9zwso9m
Inspect in browser:
  https://debug.bun.sh/#localhost:6499/4hki9zwso9m
--------------------- Bun Inspector ---------------------
Server running at http://localhost:3000/
Thread 0x1e0929300 encountered bitfit alloaction error.
Bits for page 0x107a10230 (bmalloc_marge_bitfit):
free bits: 0000000000000000000011111110000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000100000000000000000000000000000000000000011111111111111111111111110000000000000001111111111111111111111111111111111111111000000000000000000000000000111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
 end bits: 0000000000000000000100000000000000000000000000000000000001000001000000000000000000010000001000000000000000000000000000000100000100000000000000000000000000100000000000000000010000010000010000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000010000000000000000000000001000000000001000000001000000000000001000000000000000010000000100000001000000000000000001000000000000000100000100000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
                                                                                                                                                                                                                                                                                                                                                                                                                                                          ^                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
[6182] pas panic: deallocation did fail at 0x1151af000: free bit set
```

**Comment by subtleGradient** [comment]
more weirdness triggered by this:
https://github.com/subtleGradient/rsc-fun-in-bun/blob/b69b78ae63b13795c02c97aa8dd102cd0112d5b9/demo-RSC.server.tsx

```
% bun run ./demo-RSC.server.tsx        
Server running at http://localhost:3000/
Thread 0x1e0929300 encountered bitfit alloaction error.
Bits for page 0x109c7c230 (bmalloc_marge_bitfit):
free bits: 0000000000000000000011111110000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000111000000000000000000000000000000000000000000000000000000000000000000000000000001111111111100000000000000001111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
 end bits: 0000000000000000000100000000000000000000000000000000000001000001000000100000000000000001000001000001000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000001000001000000000000001000000000000001000000000000000010000000100000001000000000000000001000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000001000000000000000000000000000000000000000000000000010000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002315 | }
        ^
error: Expected scope (src.js_ast.Scope.Kind.function_args, -863356908) in input.jsx, found scope (src.js_ast.Scope.Kind.function_args, 2760)
    at input.jsx:2315:2

0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               ^                                                                                                                                     
uh-oh: 2315 | }
        ^
error: Expected scope (src.js_ast.Scope.Kind.function_args, -863356908) in input.jsx, found scope (src.js_ast.Scope.Kind.function_args, 2760)
    at input.jsx:2315:2

bun will crash now 😭😭😭
                                                                                                                                                                                                                                                                                                                                                                                                                                      
[7816] pas panic: deallocation did fail at 0x1371d4000: free bit set

----- bun meta -----
Bun v1.0.36 (40f61ebb) macOS Silicon 23.0.0
RunCommand: tsconfig transpiler_cache(4) http_server 
Builtins: "bun:main" 
Elapsed: 1630ms | User: 79ms | Sys: 32ms
RSS: 61.26MB | Peak: 61.26MB | Commit: 43.66MB | Faults: 19
----- bun meta -----

0   0x105cd4050
1   ???

Search GitHub issues https://bun.sh/issues or ask for #help in https://bun.sh/discord

thread 79249 panic: 2315 | }
        ^
error: Expected scope (src.js_ast.Scope.Kind.function_args, -863356908) in input.jsx, found scope (src.js_ast.Scope.Kind.function_args, 2760)
    at input.jsx:2315:2

zsh: abort      bun run ./demo-RSC.server.tsx
```

**Comment by Jarred-Sumner** [comment]
This is a bug in Bun.Transpiler which will be fixed in Bun v1.1.0 in #9751
